File: 2550-words-within-two-edits-of-dictionary/words-within-two-edits-of-dictionary.cpp
class Solution {
public:
    vector<string> twoEditWords(vector<string>& queries,
                                vector<string>& dictionary) {
        vector<string> res;
        for (string query : queries) {
            for (string word : dictionary) {
                int hamming_dist = 0;
                for (int i = 0; i < query.size(); i++) {
                    if (query[i] != word[i]) {
                        hamming_dist++;
                    }
                    if (hamming_dist > 2) {
                        break;
                    }
                }
                if (hamming_dist <= 2) {
                    res.push_back(query);
                    break;
                }
            }
        }
        return res;
    }
};

File: 2550-words-within-two-edits-of-dictionary/words-within-two-edits-of-dictionary.js
var twoEditWords = function (queries, dictionary) {
    const res = [];
    for (const query of queries) {
        for (const word of dictionary) {
            let hamming_dist = 0;
            for (let i = 0; i < query.length; i++) {
                if (query[i] !== word[i]) {
                    hamming_dist++;
                }
                if(hamming_dist > 2){
                    break;
                }
            }
            if (hamming_dist <= 2) {
                res.push(query);
                break;
            }
        }
    }
    return res;
};